tests/functional/conftest.py

```diff
@@ -94,7 +94,16 @@ def reset_gitlab(gl: gitlab.Gitlab) -> None:
             helpers.safe_delete(deploy_token)
         logging.info(f"Deleting project: {project.path_with_namespace!r}")
         helpers.safe_delete(project)
+
     for group in gl.groups.list():
+
+        # skip deletion of a descendant group to prevent scenarios where parent group gets deleted leaving a dangling descendant whose deletion will throw 404s.
+        if group.parent_id:
+            logging.info(
+                f"Skipping deletion of {group.full_path} as it is a descendant group and will be removed when the parent group is deleted"
+            )
+            continue
```

Comment:
yep, it's the problem of whoever wants to use this function next! After all the changes, just realised I spent more time on a function I ended up not using in my tests than on the actual feature lol :')

Reply:
actually, I had to revert this change as it started to inexplicably fail CLI functional tests.

Reply:
Hehe @Shreya-7 thanks for your patience and sorry for the rollercoaster ride here, still good to see this can affect other tests in the future (whichever way it's done with subgroups) and we'll keep this in mind. Could you maybe just squash your PR to a single commit here for brevity in the release notes? I can do it too if you've had enough :sweat_smile:

Reply:
no worries! Do we want the commit message to include the reset_gitlab fix as well or just the feature?

Reply:
There was a bit more work as there were some merge commits in between so I just force-pushed a squashed commit t o your branch @Shreya-7 ;) I'll just wait for CI to run now and otherwise LGTM!

Reply:
Thanks so much!!!